### ISSUE TITLE
Cannot Install user app

### ISSUE THREAD

**Issue by santiagoc81** [OfficeDev/microsoft-teams-apps-company-communicator#974]
Hello Team,
I am trying to do a new deploy and it fails with this error:

{"code":"InvalidTemplateDeployment","details":[{"details":[{"code":"InvalidBotData","target":"ccwfsac2023-author","message":"The Azure App Insights key is not a valid GUID"},{"code":"InvalidBotData","target":"ccwfsac2023","message":"The Azure App Insights key is not a valid GUID"}]}],"message":"The template deployment 'Microsoft.Template-20230227214006' is not valid according to the validation procedure. The tracking id is 'a5bef2df-a0f0-45a5-872f-8c3926aa8b69'. See inner errors for details."}

Can you help me?

**Comment by santiagoc81** [comment]
I changed all GUIDs and also recreated the app registrations. Same issue

**Comment by gsv022** [comment]
Hi @santiagoc81 ,
Thanks for raising the query. 

- Could you please share the screenshots showing the inputs given while creating the ARM template.

-  Use this link to generate 7 [GUID](https://www.guidgenerator.com/online-guid-generator.aspx) and paste it in the below 7 places and try validating it.

<img width="453" alt="image" src="https://user-images.githubusercontent.com/98143492/218417396-4b3b2988-e4f3-4fed-95ff-89f56ef89265.png">

**Comment by santiagoc81** [comment]
Hello
Same error, look at screenshot:
![GUIDs](https://user-images.githubusercontent.com/3265382/221905045-1cffa45d-1040-45e0-b45b-066a68269611.png)

**Comment by gsv022** [comment]
We are working on this issue and will get back to you with proper response soon

**Comment by v-jaygupta** [comment]
@santiagoc81,

We are prioritizing this issue as our highest concern, and we have tentatively identified the problem. Currently, we are testing a potential solution, and we will confirm once we have found a working fix.

**Comment by v-jaygupta** [comment]
Hi @santiagoc81 ,
We identified that the issue is not specific to the company communicator but is due to a recent update in either Bot Services or Azure template custom validation. The update is preventing the creation of a reference of App Insights Instrument key to the Microsoft Bot Service configuration. This change is impacting other templates and repositories that are dependent on it, including company communicator.
We have identified a possible workaround and are currently testing it with the highest priority. As soon as testing is complete, we will share the steps to unblock the deployment. Thank you for your patience and understanding as we work to resolve this issue.

**Comment by gsv022** [comment]
Hi @santiagoc81 ,
- It is identified that the issue is not specific to the company communicator but is evident that there is an issue with Azure ARM resources functionality.
- We have updated the [Deploy to Azure](https://portal.azure.com/?feature.customportal=false/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FOfficeDev%2Fmicrosoft-teams-company-communicator-app%2Fmain%2FDeployment%2Fazuredeploy.json) link in 2nd step of the CC Deployment Guide. The earlier faced issue "_The Azure App Insights key is not a valid GUID_" is now resolved and deployment is working as expected.
- Please try deploying the Company communicator app and let us know if you face any further issues. 
- If the deployment is failing due to role assignments, please refer to the [Troubleshooting Guide](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/wiki/Troubleshooting#4-multiple-instance-of-company-communicator-in-the-same-tenant-action-items).

**Comment by gsv022** [comment]
@santiagoc81 ,Please let us know the status if the issue persists. Otherwise we are good to close out the issue.

**Comment by gsv022** [comment]
@santiagoc81 ,We are closing this request as we have provided the inputs for the query raised. If you face any issue with company communicator please raise a new issue for further assistance.
